### PR TITLE
Use System.getProperty for ttl values found using legacy sun properties

### PR DESCRIPTION
**Pull request by lorenzhawkes**
The ordinary java implementation uses Security.getProperty for:
networkaddress.cache.ttl
networkaddress.cache.negative.ttl

and then uses System.getProperty for:
sun.net.inetaddr.ttl
sun.net.inetaddr.negative.ttl

Change the InetAddressDnsResolver to follow the same pattern to allow
setting sun.net.inetaddr.* properties from the command line.

### CODE REVIEW

**Comment by lorenzhawkes** [comment]
Fixes issue #24188

**Comment by ktoso** [comment]
OK TO TEST

File: akka-actor-tests/src/test/scala/akka/io/InetAddressDnsResolverSpec.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package akka.io
```
Comment (ktoso) [inline akka-actor-tests/src/test/scala/akka/io/InetAddressDnsResolverSpec.scala]: Has to have header as all files in akka:

```
 /**
  * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
 */
```

File: akka-actor-tests/src/test/scala/akka/io/InetAddressDnsResolverSpec.scala [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    akka.io.dns.inet-address.positive-ttl = default
+    akka.io.dns.inet-address.negative-ttl = default
+    akka.actor.serialize-creators = on
+    """) { thisSpecs ⇒
+
+  "The DNS resolver default ttl's" must {
+    "use the default value for positive caching if it is not overridden" in {
+      withNewSecurityProperty("networkaddress.cache.ttl", "") {
+        withNewSystemProperty("sun.net.inetaddr.ttl", "") {
+          dnsResolver.positiveTtl shouldBe secondsToMillis(30)
```
Comment (ktoso) [inline akka-actor-tests/src/test/scala/akka/io/InetAddressDnsResolverSpec.scala]: could fail depending on JVM version though, right?

Comment (lorenzhawkes) [reply akka-actor-tests/src/test/scala/akka/io/InetAddressDnsResolverSpec.scala]: I don't think so. Although the assumption of the test is based on the Java 8 implementation, the Java 9 version remains with the same logic (although implementation has changed a little) 

http://hg.openjdk.java.net/jdk9/jdk9/jdk/file/19cbbf152335/src/share/classes/sun/net/InetAddressCachePolicy.java

**Review by ktoso — APPROVED** [review]
LGTM, the Security there was likely my typo.
Please note that we did not copy the impl but reimplemented behaviour.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**